package/MDAnalysis/core/groups.py

```diff
@@ -601,6 +602,35 @@ def dimensions(self):
     def dimensions(self, dimensions):
         self.universe.trajectory.ts.dimensions = dimensions
 
+    @property
+    @cached('isunique')
+    def isunique(self):
+        """Boolean indicating whether all members in the group are unique, i.e.,
+        the group contains no duplicates.
+
+        Examples
+        --------
+
+           >>> ag = u.atoms[[2, 1, 2, 2, 1, 0]]
+           >>> ag
+           <AtomGroup with 6 atoms>
+           >>> ag.isunique
+           False
+           >>> ag2 = ag.unique
+           >>> ag2
+           <AtomGroup with 3 atoms>
+           >>> ag2.isunique
+           True
+
+        .. versionadded:: 0.19.0
+        """
+        if len(self) <= 1:
+            return True
+        # Check indices for uniqueness:
+        s_ix = np.sort(self._ix)
+        mask = s_ix[1:] != s_ix[:-1]
+        return mask[mask.argmin()] # faster than mask.min()
```

Comment:
This is pretty tough to read, can you add a comment that this is equivalent to `not any(mask)`, ie a duplicate was found